packages/code-du-travail-ui/src/Toast/index.js

```diff
@@ -0,0 +1,186 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled, { css } from "styled-components";
+import IconWarning from "react-feather/dist/icons/alert-triangle";
+import IconInfo from "react-feather/dist/icons/info";
+import IconSuccess from "react-feather/dist/icons/check";
+import IconClose from "react-feather/dist/icons/x";
+import ToggleButton from "../ToggleButton";
+import { box, colors, fonts, spacing } from "../theme";
+import { fromTop, fromRight, fromBottom, fromLeft } from "../keyframes";
+
+/**
+ * Toast element used to show a message with a specific background
+ */
+class Toast extends Component {
+  constructor(props) {
+    super(props);
+    this.timer = null;
+  }
+
+  componentDidMount() {
+    const { timeout } = this.props;
+
+    if (timeout) {
+      this.timer = setTimeout(() => {
+        this.props.onRemove();
+      }, timeout);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+  }
+
+  render() {
+    const { animate, onRemove, type, wide } = this.props;
+    let Icon = IconInfo;
+    if (type === "warning") Icon = IconWarning;
+    if (type === "success") Icon = IconSuccess;
+
+    return (
+      <StyledToast type={type} animate={animate} wide={wide}>
+        <IconWrapper type={type}>
+          <Icon />
+        </IconWrapper>
+        <Content role="alert">{this.props.children}</Content>
+        {onRemove ? (
+          <ButtonWrapper>
+            <ToggleButton variant="icon" onClick={onRemove}>
+              <IconClose />
+            </ToggleButton>
+          </ButtonWrapper>
+        ) : null}
+      </StyledToast>
+    );
+  }
+}
+
+Toast.propTypes = {
+  /**
+   * Type of toast to show
+   */
+  type: PropTypes.oneOf(["warning", "success", "info"]),
```

Comment:
I wonder if we should use flavors here ?

Reply:
My secret goal is to shoot the flavor utils 🤓